### DISCUSSION TITLE
Electron 应用插件市场问题咨询

### DISCUSSION THREAD

**Discussion by PerfectPan** [opensumi/core#2710] [Q&A] [answered]
如果我想插件市场展示的列表为本地的插件文件，这个有办法可以做到么？工具有私有化部署的场景，直接用插件市场比较麻烦，想先简单内置默认若干插件 + 提供一些插件给用户可选安装。

而且本身也有只暴露若干插件的诉求，而不是全集。

我看支付宝开发者工具扩展市场展示的插件列表都是小程序相关的，有些乱七八糟的插件都没了，这是自己单独另外部署了插件市场？
![image](https://github.com/opensumi/core/assets/24316656/393630c3-1278-460c-9f52-276ecf49f0e0)

**Comment by erha19** [comment, marked as answer]
@PerfectPan 

> 如果我想插件市场展示的列表为本地的插件文件，这个有办法可以做到么？工具有私有化部署的场景，直接用插件市场比较麻烦，想先简单内置默认若干插件 + 提供一些插件给用户可选安装。

这个是可实现的，不过需要你自行修改插件市场模块相关代码，将安装插件的来源及下载逻辑进行重新实现。

支付宝小程序开发工具使用的是我们内部一个中心化的插件市场服务，能根据不同客户端/团队维度的插件管理能力去提供插件。

**Reply by PerfectPan** [reply]
> 这个是可实现的，不过需要你自行修改插件市场模块相关代码，将安装插件的来源及下载逻辑进行重新实现。

哦哦，请问这个是说直接去改 npm 包的代码么？还是有别的可覆盖的方式。

> 支付宝小程序开发工具使用的是我们内部一个中心化的插件市场服务，能根据不同客户端/团队维度的插件管理能力去提供插件。

理解了

**Reply by erha19** [reply]
@PerfectPan 这个需要你们评估一下实现的成本，修改不复杂可以直接通过服务覆盖去实现，复杂的话还是建议另外起一个包进行实现，可以复用 `@opensumi/ide-extension-manager` 内原有的服务实现，把下载和展示逻辑相关的代码进行重写，引入时也引入你们重写实现后的模块。

如：https://github.com/opensumi/opensumi-module-samples 这里的一些案例结构

**Reply by PerfectPan** [reply]
> @PerfectPan 这个需要你们评估一下实现的成本，修改不复杂可以直接通过服务覆盖去实现，复杂的话还是建议另外起一个包进行实现，可以复用 `@opensumi/ide-extension-manager` 内原有的服务实现，把下载和展示逻辑相关的代码进行重写，引入时也引入你们重写实现后的模块。
> 
> 如：https://github.com/opensumi/opensumi-module-samples 这里的一些案例结构

好的，感谢🙏

**Reply by bytemain** [reply]
https://marketplace.opentrs.cn/square 现在也支持团队功能了，你们可以试试创建自己的团队

**Reply by PerfectPan** [reply]
> https://marketplace.opentrs.cn/square 现在也支持团队功能了，你们可以试试创建自己的团队

好的，我一并研究下
